libopae/pluginmgr.c

```diff
@@ -79,20 +79,49 @@ STATIC int opae_plugin_mgr_plugin_count;
 #define CFG_PATHS 4
 static const char *_opae_cfg_files[CFG_PATHS] = {
 	"/etc/opae.cfg",
-	"~/.config/opae/opae.cfg",
-	"~/.local/opae/opae.cfg",
-	"~/.local/opae.cfg"
+	"$HOME/.config/opae/opae.cfg",
+	"$HOME/.local/opae/opae.cfg",
+	"$HOME/.local/opae.cfg"
 };
 
+STATIC void resolve_file_name(char *dst, const char *src)
+{
+	char src_cpy[PATH_MAX];
+	char *ptok = src_cpy;
+	char *pstr = src_cpy;
+	char *pdst = dst;
+	size_t len = strlen(src);
+	size_t len_cpy = len;
+
+	strncpy_s(src_cpy, PATH_MAX, src, len);
+
+	while(ptok && len_cpy) {
+		*pdst++ = '/';
+		ptok = strtok_s(NULL, &len_cpy, "/", &pstr);
+		if (ptok[0] == '$') {
+			strncpy_s(pdst, len, getenv(ptok+1), len);
```

Comment:
Probably should check whether getenv returns NULL and provide an error return code.